app/app/src/main/kotlin/com/hedvig/app/feature/chat/data/ChatRepositoryNew.kt

```diff
@@ -0,0 +1,120 @@
+package com.hedvig.app.feature.chat.data
+
+import arrow.core.Either
+import arrow.core.raise.either
+import com.apollographql.apollo3.ApolloClient
+import com.hedvig.android.apollo.safeExecute
+import com.hedvig.android.apollo.toEither
+import com.hedvig.android.core.common.ErrorMessage
+import com.hedvig.android.logger.LogPriority
+import com.hedvig.android.logger.logcat
+import kotlinx.datetime.Instant
+import octopus.ChatMessagesQuery
+import octopus.ChatSendFileMutation
+import octopus.ChatSendMessageMutation
+import octopus.fragment.MessageFragment
+import octopus.type.ChatMessageFileInput
+import octopus.type.ChatMessageSender
+import octopus.type.ChatMessageTextInput
+
+interface ChatRepositoryNew {
+  suspend fun fetchChatMessages(until: Instant? = null): Either<ErrorMessage, ChatMessagesResult>
+  suspend fun sendFile(uploadUrl: String): Either<ErrorMessage, ChatMessageResult>
+  suspend fun sendMessage(text: String): Either<ErrorMessage, ChatMessageResult>
+}
+
+class ChatRepositoryNewImpl(
+  private val apolloClientOctopus: ApolloClient,
+) {
+  suspend fun fetchChatMessages(until: Instant? = null) = either {
+    val result = apolloClientOctopus.query(ChatMessagesQuery(until))
+      .safeExecute()
+      .toEither(::ErrorMessage)
+      .bind()
+
+    ChatMessagesResult(
+      messages = result.chat.messages.mapNotNull { it.toMessage() },
+      nextUntil = result.chat.nextUntil,
+      hasNext = result.chat.hasNext,
+    )
+  }
+
+  suspend fun sendFile(uploadUrl: String) = either {
```

Comment:
What will `uploadUrl` be here?
I suppose at some point we will need to convert the URI from the location in the device into a `File` and then into an apollo `Upload`, kinda how we do here https://github.com/HedvigInsurance/android/blob/69ef86b42ccc12b14c1ce6145d8a63815cd68c44/app/feature/feature-chat/src/main/kotlin/com/hedvig/android/feature/chat/ChatRepository.kt#L140

I wonder what the easiest work with type would be for the repository function to accept. Maybe it's String, while also being the least type-safe of course.